command/agent/template/template.go

```diff
@@ -246,31 +246,23 @@ func (ts *Server) Run(ctx context.Context, incoming chan string, templates []*ct
 				ts.runner.Stop()
 				return nil
 			}
-		default:
-			// We are using default instead of a new case block to prioritize the case where <-incoming has a new value over
-			// receiving an error message from the consul-template server
-			select {
-			case err := <-ts.runner.ServerErrCh:
-				var responseError *api.ResponseError
-				ok := errors.As(err, &responseError)
-				if !ok {
-					ts.logger.Error("template server: could not extract error response")
-					continue
-				}
-				if responseError.StatusCode == 403 && strings.Contains(responseError.Error(), logical.ErrInvalidToken.Error()) && !tokenRenewalInProgress.Load() {
-					ts.logger.Info("template server: received invalid token error")
-
-					// Drain the error channel before sending a new error
-					select {
-					case <-invalidTokenCh:
-					default:
-					}
-					invalidTokenCh <- err
-				}
-			default:
+		case err := <-ts.runner.ServerErrCh:
+			var responseError *api.ResponseError
+			ok := errors.As(err, &responseError)
```

Comment:
I wonder if this is a possible scenario - 

- IncomingToken receives a new token at the same time template sends an error back to ServerErrCh
- We reauthenticate first by honoring the ServerErrCh select first. Now IncomingCh has two values in the channel
- We try using the first token in IncomingCh but there is an error. Another error is sent to ServerErrCh
- Again the SeverErrCh is honored first and we reauthenticate. 


We are now stuck in a loop where we always honor the token one behind valid token.